src/ensembl/src/shared/components/loading-button/LoadingButton.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useRef, ReactNode, useEffect } from 'react';
+import { of, from, merge, timer, combineLatest, Subscription } from 'rxjs';
+import { tap, mergeMap, delay, map, take, catchError } from 'rxjs/operators';
+
+import ControlledLoadingButton from './ControlledLoadingButton';
+
+import { LoadingState } from 'src/shared/types/loading-state';
+
+type LoadingButtonProps = {
+  onClick: () => Promise<unknown>;
+  onSuccess?: (x?: unknown) => void;
+  onError?: (x?: unknown) => void;
+  isDisabled?: boolean;
+  className?: string;
+  children: ReactNode;
+};
+
+type RequestHandlerParams = {
+  loader: () => Promise<unknown>;
+  onSuccess?: (x?: unknown) => void;
+  onError?: (x?: unknown) => void;
+};
+
+const MINIMUM_SPINNER_TIME = 1000; // 1 second
+const SUCCESS_INDICATOR_TIME = 2000; // 2 seconds
+
+const getLoadingState$ = (params: RequestHandlerParams) => {
+  const loadingStart$ = of(LoadingState.LOADING);
+  const request$ = combineLatest([
+    timer(MINIMUM_SPINNER_TIME).pipe(take(1)),
+    from(params.loader()).pipe(
+      map((result) => ({
+        status: LoadingState.SUCCESS as const,
+        result
+      })),
+      catchError((error) =>
+        of({
+          status: LoadingState.ERROR as const,
+          error
+        })
+      )
+    )
+  ]).pipe(
+    tap(([, result]) => {
+      if (result.status === LoadingState.SUCCESS) {
+        params?.onSuccess?.(result.result);
+      } else {
+        params?.onError?.(result.error);
+      }
+    }),
+    mergeMap(([, result]) => {
+      return merge(of(result.status), returnToIniital$);
```

Comment:
Is this deprecated?
<img width="693" alt="Screenshot 2021-03-16 at 21 19 38" src="https://user-images.githubusercontent.com/3085815/111381263-5f318380-869d-11eb-9411-4dfc7b7661b2.png">

Reply:
There are two merges in the rxjs library 😄 The one that's an operator is [deprecated](https://rxjs-dev.firebaseapp.com/api/operators/merge), but the one that is a creator of observables [isn't](https://rxjs-dev.firebaseapp.com/api/index/function/merge). The docs [specifically say](https://rxjs-dev.firebaseapp.com/api/operators/merge) that the first merge is deprecated in favour of the second one. Not sure what the popup is trying to say. Maybe it's picking up some of the overloads?

Try importing a merge from `rxjs/operators` for comparison, and check what a real deprecation warning looks like. I think it will look differently. VS Code was underlining that merge with a yellow line and was generally fussy about it.